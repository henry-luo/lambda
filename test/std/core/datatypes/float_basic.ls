// Test: Floating-Point Basic Operations
// Category: core/datatypes
// Type: positive
// Expected: 1

// Test basic floating-point operations
let pi = 3.14159
let e = 2.71828

// Basic arithmetic
let sum = pi + e
let diff = pi - e
let product = pi * e
let quotient = pi / e

// Mixed integer and float operations
let mixed = 2 * pi + 1
let mixed_div = 10 / 3.0  // Should be 3.333...

// Scientific notation
let avogadro = 6.022e23
let small_num = 1.6e-19

// Edge cases
let inf = 1.0 / 0.0  // Positive infinity
let neg_inf = -1.0 / 0.0  // Negative infinity
let nan = 0.0 / 0.0  // Not a Number

// Final check
1
