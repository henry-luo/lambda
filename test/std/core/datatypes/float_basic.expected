3.14159
2.71828
5.85987
0.42331
8.53973
1.15573
7.28318
3.33333
6.022e+23
1.6e-19
Infinity
-Infinity
NaN
